### DISCUSSION TITLE
API Application with "Basic authentication type"

### DISCUSSION THREAD

**Discussion by kryptn36** [zitadel/zitadel#3832] [Q&A] [answered]
How does the flow for an application mentioned in the title work exactly? Seems like it isn't documented yet. A curl example would be helpful.

**Comment by livio-a** [comment, marked as answer]
Hey @kryptn36 

API applications in ZITADEL correspond to OAuth2 Resource Server. So you can mainly use them to check the received access token by calling the introspection endpoint: https://docs.zitadel.com/docs/apis/openidoauth/endpoints#introspection_endpoint
The Basic Auth variant essentially looks like this (token is the access token you received from the client):
```bash
curl --request POST \
  --url {your_domain}/oauth/v2/introspect \
  --header 'Content-Type: application/x-www-form-urlencoded' \
  --header 'Authorization: Basic <your basic auth header>' \
  --data token=VjVxyCZmRmWYqd3_F5db9Pb9mHR5fqzhn...
```
Please check the docs on how to build the Basic Auth Header correctly for an OAuth2 call: https://docs.zitadel.com/docs/apis/openidoauth/authn-methods#client-secret-basic

I'll add the example above to the docs shortly.

I hope I got your question right and the above helps you. If not just let me know.
And let me know if you need further information about the overall process (OAuth2 / OIDC applications and APIs).
